src/gaUtility.cpp

```diff
@@ -0,0 +1,23 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// This is a simple example of exporting a C++ function to R. You can
+// source this function into an R session using the Rcpp::sourceCpp 
+// function (or via the Source button on the editor toolbar). Learn
+// more about Rcpp at:
+//
+//   http://www.rcpp.org/
+//   http://adv-r.had.co.nz/Rcpp.html
+//   http://gallery.rcpp.org/
+//
+
+// [[Rcpp::export]]
+
+
+NumericVector gaUtility_rcpp(int bGA, int aGA, NumericVector GA) {
+  
+  NumericVector GA_pow = pow(GA, aGA);
+  NumericVector output = rep((bGA * GA_pow) * -1, 3);
```

Comment:
It's possible that doing this manually in a loop per element may be faster, because there are 4 NumericVectors being constructed here just to get one output NumericVector.
```suggestion
  NumericVector output = rep(0, 3 * GA.length());
  for (std::size_t ix = 0; ix < GA.length(), ++ix) {
    auto number = -bGA * std::pow(GA[ix], aGA);
    output[ix] = number;
    output[ix + GA.length()] = number;
    output[ix + 2 * GA.length()] = number;
  }
  NumericVector output = rep((bGA * GA_pow) * -1, 3);
```

Also note that you're multiplying an integer (`bGA`) with a floating point vector; that conversion may also be a micro-slowdown.

Reply:
By the way, note that in the suggestion above I use `std::size_t` for the index variable type in the for loop. Opinions on using this as a default indexing type differ, but I'm personally of the opinion that using an unsigned type for indices is better than using signed types (e.g. `int`), just because it better conveys the semantics of an index type: negative indices don't technically exist for arrays (even though they of course make convenient short-hand notation in e.g. Python indexing).

It may be that using it as I do in this suggestion produces a warning in this case ("comparison between signed and unsigned ..."), I'm not sure if `GA.length()` actually returns a signed or unsigned integer. If signed, I would personally cast the length to unsigned, but I would understand if people made the choice then of keeping things signed, just to reduce the clutter :)

Reply:
I'm not sure why, but this version throws an error for 'NumericVector output = rep(0, 3 * GA.length());' --> no viable conversion from sugar::Rep_Single<int> to Rcpp::NumericVector (aka 'Vector<14>') -- I haven't maneged to get a workaround

Reply:
Hmm, I can take a look!

Reply:
This code should work I think. 

`
NumericVector output(3 * GA.length());
  
  for (int ix = 0; ix < GA.length(); ++ix) {
    auto number = -bGA * std::pow(GA[ix], aGA);
    output[ix] = number;
    output[ix + GA.length()] = number;
    output[ix + 2 * GA.length()] = number;
  }
`

I benchmarked it against the previous version and the R version, and it's slightly faster than both this way!

Reply:
Nice, great job!